byzcoin/contracts/coins.go

```diff
@@ -95,6 +95,10 @@ func (c *contractCoin) Invoke(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Instru
 			err = errors.New("argument \"coins\" is missing")
 			return
 		}
+		if len(coinsBuf) != 8 {
```

Comment:
Nitpick: is this the correct place to improve the coins behaviour?

Reply:
Where else would it be? I agree some sort of nice "coin" abstract type would be fun to write, but it seems too big of an investment for this little problem.